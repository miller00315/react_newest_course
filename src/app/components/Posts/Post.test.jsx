import { render, screen } from '@testing-library/react';
import {Posts} from '.';

const mockedPosts = [{
        title: 'testTitle1',
        body: 'testBody1',
        id: 1,
        cover: 'img/img1.png'
    },
    {
        title: 'testTitle2',
        body: 'testBody2',
        id: 2,
        cover: 'img/img2.png'
    },
    {
        title: 'testTitle3',
        body: 'testBody3',
        id: 3,
        cover: 'img/img3.png'
    },
];

describe('<Post />', () => {
    it('Should render posts', () => {
        render(<Posts posts={mockedPosts}/>);

        expect(screen.getAllByRole('heading', {name: /title/i})).toHaveLength(3);
        expect(screen.getAllByRole('img', {name: /title/i})).toHaveLength(3);
        expect(screen.getAllByText(/body/i)).toHaveLength(3);
    });

    it('Should match snapshot', () => {
        const container = render(<Posts posts={mockedPosts}/>);

        expect(container.firstChild).toMatchSnapshot();
    });
});