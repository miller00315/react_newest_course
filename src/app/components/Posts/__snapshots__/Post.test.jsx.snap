// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> Should match snapshot 1`] = `undefined`;
